src/fs-helpers.ts

```diff
@@ -99,24 +136,52 @@ function projectInternal(cwd: string) {
     function dir(path: string, cb?: (dir: DirectoryApi) => void) {
       return createDirectory(Path.join(dirPath, path), cb);
     }
+    function readFrom(realFsDirPath: string, targetPath?: string, ignoredPaths?: string[]): void {
+      const targetDir = targetPath ? dir(targetPath) : _dir;
+      readFromFsIntoFixture(realFsDirPath, targetDir);
+    }
+    function getFile(path: string): BaseFile | undefined {
+      const filePath = Path.join(dirPath, path);
+      // Search for most recently appended in case the same file was written multiple times,
+      // because we do not dedupe on write.
+      // A bit of a hack.
+      for(let i = files.length - 1; i >= 0; i--) {
+        const file = files[i];
+        if (file.path === filePath) return file;
+      }
+    }
+    function getJsonFile(path: string): JsonFile<any> | undefined {
+      const found = getFile(path);
+      if((found as JsonFile).type !== 'json') throw new Error(`Found file in fixture, but it is type ${(found as File).type} instead of json.`);
```

Comment:
If we don't find the file, this will throw rather tahn returning `undefined`.
```suggestion
      if(found && (found as JsonFile).type !== 'json') throw new Error(`Found file in fixture, but it is type ${(found as File).type} instead of json.`);
```

Reply:
good catch, fixed